server/apps/main/tests/test_views.py

```diff
@@ -304,6 +304,127 @@ def test_share_exp_submit_new_experience(self):
         # Check that there is a redirect after
         assert response.status_code == 200
 
+    @vcr.use_cassette(
+        "server/apps/main/tests/fixtures/share_experience.yaml",
+        record_mode="none",
+        filter_query_parameters=["access_token"],
+        match_on=["path"],
+    )
+    def test_share_exp_redirect_confirmation(self):
+        """
+        Test that the user is redirected to the user profile
+        given the profile is incomplete and the experience will be
+        shared publicly and for research
+
+        logic:
+            request.user.is_authenticated == True
+            request.method == "POST"
+            form.is_valid() == True
+            viewable == True
+            research == True
+            user.autistic_identification == ''
+        """
+        c = Client()
+        c.force_login(self.user_a)
+
+        user_a_stories_before = len(
+            PublicExperience.objects.filter(open_humans_member=self.oh_a)
+        )
+
+        response = c.post(
+            "/main/share_exp/",
+            {
+                "experience_text": "Here is some experience text",
+                "difference_text": "Here is some difference text",
+                "title_text": "A new story added",
+                "viewable": "False",
+                "research": "False",
+                "open_humans_member": self.oh_a,
+            },
+            follow=True,
+        )
+
+        user_a_stories_after = len(
+            PublicExperience.objects.filter(open_humans_member=self.oh_a)
+        )
+
+        pe_db_after = PublicExperience.objects.filter(title_text="A new story added")
+
+        # Check that a story with the title now exists in the database
+        assert len(pe_db_after) == 0
```

Comment:
Maybe I'm having a brainfart, but shouldn't this be `len==1` if the story exists?

Reply:
Nevermind, I guess I understood now, It's 0 for both as the experience is not public, it's just that the comments need to be adjusted to reflect what we actually test for :)